packages/fuel-indexer/src/executor.rs

```diff
@@ -156,94 +190,84 @@ pub fn run_executor<T: 'static + Executor + Send + Sync>(
+                        "Indexer({indexer_uid}) failed after too many retries, giving up. <('.')>"
                     );
                     break;
                 }
-            }
 
-            if cursor.is_none() {
-                num_empty_block_reqs += 1;
+                if let IndexerError::SqlxError(sqlx::Error::Database(inner)) = e {
+                    // TODO: https://github.com/FuelLabs/fuel-indexer/issues/1093
+                    if inner.constraint().is_some() {
+                        // Just bump the cursor and keep going. These errors do not count towards `INDEXER_FAILED_CALLS`
+                        warn!("Constraint violation. This is not a retry-able error. Continuing...");
+                        cursor = next_cursor;
+                        continue;
+                    }
+                }
 
-                info!(
-                    "Indexer({indexer_uid}) has no new blocks to process, sleeping. zzZZ"
-                );
-                sleep(Duration::from_secs(DELAY_FOR_EMPTY_PAGE)).await;
+                // If we get here, this must be an error that allows us to retry.
+                warn!("Indexer({indexer_uid}) retrying handler after {retry_count}/{INDEXER_FAILED_CALLS} failed attempts.");
 
-                if num_empty_block_reqs == max_empty_block_reqs {
-                    error!("No blocks being produced, Indexer({indexer_uid}) giving up. <('.')>");
-                    break;
-                }
-            } else {
-                next_cursor = cursor;
-                num_empty_block_reqs = 0;
+                retry_count += 1;
+
+                // Since there was some type of error, we're gonna call `retrieve_blocks_from_node` again,
+                // with our same cursor.
+                continue;
             }
 
+            // If we get a non-empty response, we reset the counter.
+            num_empty_block_reqs = 0;
+
+            // If we make it this far, we always go to the next page.
+            cursor = next_cursor;
+
+            // Again, check if something else has signaled that this indexer should stop, then stop.
             if kill_switch.load(Ordering::SeqCst) {
                 info!("Kill switch flipped, stopping Indexer({indexer_uid}). <('.')>");
                 break;
             }
 
+            // Since we had successful call, we reset the retry count.
             retry_count = 0;
         }
     }
 }
 
 /// Retrieve blocks from a client node.
 ///
-// This was abstracted out of `run_executor` in order to allow for
-// use in the benchmarking suite to give consistent timings.
+/// This was abstracted out of `run_executor` in order to allow for use in the benchmarking suite
+/// to give consistent timings.
+///
+/// If there is an issue fetching blocks, we return an empty cursor and an empty list of blocks, then
+/// run_executor will determine whether or not we should sleep for a bit and try again.
```

Comment:
This can be an internal comment as the reasoning probably doesn't need to be included in the public rustdocs, just my two cents.

Reply:
@deekerno 
- So previously I've agreed that these should be internal comments.
- However, after looking over other projects (namely tokio and sqlx) I see they have a _ton_ of rustdoc comments
  - They have tons of rustdoc comments that we would definitely consider "internal comments"
  - Which makes sense, so devs reading on crates.io don't have to open the source code to see the maintainer's important comments
- So yes, in the past I've agreed that these were internal (and have made such changes to my PRs)
- However, now I'm gonna push back a bit ;) and say we can be liberal with the rustdoc comments (given that other high visibility projects do the same)

Reply:
I'm always down to look like the popular projects. 😄